.github/workflows/main.yml

```diff
@@ -0,0 +1,211 @@
+
+      - name: Restore Yarn packages
+        id: yarn-restore
+        uses: actions/cache/restore@v3
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      # Restore node modules only if the SHA is perfect; to avoid possibility of corruption.
+      - name: Restore node_modules
+        id: node_modules-restore
+        uses: actions/cache/restore@v3
+        with:
+          path: ./node_modules
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('yarn.lock') }}
+
+      - name: Yarn install
+        if: steps.node_modules-restore.outputs.cache-hit != 'true'
+        run: yarn install
+
+      - name: Save Yarn packages
+        if: steps.node_modules-restore.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v3
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
+
+      - name: Save node_modules
+        if: steps.node_modules-restore.outputs.cache-hit != 'true'
+        uses: actions/cache/save@v3
+        with:
+          path: ./node_modules
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('yarn.lock') }}
+
+  test_components:
+    runs-on: ubuntu-latest
+
+    needs: [yarn_install]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          # The storybook runner is going to do some Git operations that rely on access to the main branch.
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Setup Node
+        uses: getappmap/appmap-js@actions/setup-node/v1
+
+      - name: Build
+        run: |
+          yarn workspaces foreach -v -t --exclude root --exclude @appland/appmap --exclude @appland/scanner run build
+
+      - name: Test
+        run: |
+          cd packages/components
+          yarn run lint
+          yarn run test
+
+  test_chromatic:
+    runs-on: ubuntu-latest
+
+    if: github.event_name == 'pull_request'
+
+    needs: [yarn_install]
+
+    env:
+      CHROMATIC_PROJECT_TOKEN: ${{ secrets.CHROMATIC_PROJECT_TOKEN }}
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Node
+        uses: getappmap/appmap-js@actions/setup-node/v1
+
+      - name: Build
+        run: |
+          yarn workspaces foreach -v -t --exclude root --exclude @appland/appmap --exclude @appland/scanner run build
+
+      - name: Test
+        run: |
+          cd packages/components
+          yarn run chromatic
+
+  test_cli:
+    runs-on: ubuntu-latest
+
+    needs: [yarn_install]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: getappmap/appmap-js@actions/setup-node/v1
+
+      - name: Build
+        run: yarn run build
```

Comment:
Does it really need all the packages? `scanner` shouldn't be necessary, for one...

Reply:
You'll find lines like this one in the CLI:

```
import { scan } from '@appland/scanner';
```